pkg/asset/installconfig/aws/route53.go

```diff
@@ -156,43 +155,25 @@ func GetR53ClientCfg(sess *awss.Session, roleARN string) *aws.Config {
 	return &aws.Config{Credentials: creds}
 }
 
-// CreateOrUpdateRecord Creates or Updates the Route53 Record for the cluster endpoint.
-func (c *Client) CreateOrUpdateRecord(ctx context.Context, ic *types.InstallConfig, target string, intTarget string, phzID string, aliasZoneID string) error {
-	useCNAME := cnameRegions.Has(ic.AWS.Region)
-
-	apiName := fmt.Sprintf("api.%s.", ic.ClusterDomain())
-	apiIntName := fmt.Sprintf("api-int.%s.", ic.ClusterDomain())
-
-	// Create api record in public zone
-	if ic.Publish == types.ExternalPublishingStrategy {
-		zone, err := c.GetBaseDomain(ic.BaseDomain)
-		if err != nil {
-			return err
-		}
-
-		svc := route53.New(c.ssn) // we dont want to assume role here
-		if _, err := createRecord(ctx, svc, aws.StringValue(zone.Id), apiName, target, aliasZoneID, useCNAME); err != nil {
-			return fmt.Errorf("failed to create records for api: %w", err)
-		}
-		logrus.Debugln("Created public API record in public zone")
-	}
-
-	// Create service with assumed role for PHZ
-	svc := route53.New(c.ssn, GetR53ClientCfg(c.ssn, ic.AWS.HostedZoneRole))
+// CreateRecordInput collects information for creating a record.
+type CreateRecordInput struct {
+	Name           string
+	Region         string
+	DNSTarget      string
+	ZoneID         string
+	AliasZoneID    string
+	HostedZoneRole string
+}
 
-	// Create api record in private zone
-	if _, err := createRecord(ctx, svc, phzID, apiName, intTarget, aliasZoneID, useCNAME); err != nil {
-		return fmt.Errorf("failed to create records for api: %w", err)
-	}
-	logrus.Debugln("Created public API record in private zone")
+// CreateOrUpdateRecord Creates or Updates the Route53 Record for the cluster endpoint.
+func (c *Client) CreateOrUpdateRecord(ctx context.Context, in *CreateRecordInput) error {
+	useCNAME := cnameRegions.Has(in.Region)
 
-	// Create api-int record in private zone
-	if _, err := createRecord(ctx, svc, phzID, apiIntName, intTarget, aliasZoneID, useCNAME); err != nil {
-		return fmt.Errorf("failed to create records for api-int: %w", err)
-	}
-	logrus.Debugln("Created private API record in private zone")
+	// Create service with assumed role, if set
+	svc := route53.New(c.ssn, GetR53ClientCfg(c.ssn, in.HostedZoneRole))
 
-	return nil
+	_, err := createRecord(ctx, svc, in.ZoneID, in.Name, in.DNSTarget, in.AliasZoneID, useCNAME)
+	return err
 }
 
 func createRecord(ctx context.Context, client *route53.Route53, zoneID, name, dnsName, aliasZoneID string, useCNAME bool) (*route53.ChangeInfo, error) {
```

Comment:
What about passing in `CreateRecordInput` to this method too?

Reply:
Then I'd rather incorporate this function into the exported `CreateOrUpdateRecord`

Reply:
Done.